pyproject.toml

```
@@ -20,7 +20,7 @@ classifiers = [
   "Programming Language :: Python :: Implementation :: PyPy",
 ]
 dependencies = [
-  "pydantic~=1.10",
+  "pydantic~=2.5",
```

```suggestion
  "pydantic>=2.5",
```

You'll probably want to wait for someone from DBT to approve but it would be great to have a looser constraint here and adding an upper bound when/if it becomes necessary.

---

Hi. Yes I will let them say what to do, as I only followed how it was done before.

---

Howdy! I actually think we should stick with the `~=` specification. Pydantic follows [semantic versioning](https://semver.org/) pretty well, at least I believe it does. As such we should expect a Pydantic 3 to be breaking. It's also generally less painful after the fact to loosen the top end as compared to having to tighten it. 

 I have a separate question though, is there a specific reason to use `~=2.5` instead of `~=2.0` here? Are we using any 2.5 specific functionality in this change that requires 2.5 to be the minimum?

---

No specific reason. I probably should not have done that.

I will change it tomorrow and do some tests on lower versions than 2.5, just to make sure nothing breaks.

---

I confirm that tests pass even with pydantic `==2.0.0`, so no need for `~=2.5`. Just put it back to `~=2.0`.

@QMalcolm the floor is yours.